lib/elsa/supervisor.ex

```diff
@@ -191,16 +208,22 @@ defmodule Elsa.Supervisor do
      initializer: {Elsa.Consumer.Worker.Initializer, :init, [consumer_args]}}
   end
 
-  defp start_producer(_registry, nil), do: []
+  defp producer_spec(registry, args) do
+    initializer =
+      case Keyword.take(args, [:topic, :config]) do
+        [] -> nil
+        init_args -> {Elsa.Producer.Initializer, :init, [registry, init_args]}
```

Comment:
if `topic` is required this will technically blow up a little further downstream if you accidentally passed `[config: %{stuff}]` on its own, right? Not sure we should worry too much about that so long as the error message will bubble up that indicates the user missed the required `:topic` key

Reply:
That's where it would blow up today, so I didn't figure it was worth changing.

Reply:
Fair enough